store/src/store.rs

```diff
@@ -142,9 +169,27 @@ impl<T: KeyValueDB> ChainStore for ChainKVStore<T> {
         })
     }
 
-    fn get_header(&self, h: &H256) -> Option<Header> {
-        self.get(COLUMN_BLOCK_HEADER, h.as_bytes())
-            .map(|ref raw| unsafe { Header::from_bytes_with_hash_unchecked(raw, h.to_owned()) })
+    fn get_header(&self, hash: &H256) -> Option<Header> {
+        let mut header_cache_unlocked = self
+            .header_cache
+            .lock()
+            .expect("poisoned header cache lock");
```

Comment:
**Insignificant**

IMHO, I think `try_lock(..)` is ok, and the `expect(..)` can be removed.

Try locking the cache:
- Success: fetch data from the cache.
- Failure: fetch data from the database.

No wait, no panic.